From 0ba792563e528dc66f22188b5e65954b487cb289 Mon Sep 17 00:00:00 2001
From: Patrik Jakobsson <pjakobsson@suse.de>
Date: Mon, 8 May 2017 12:27:00 +0200
Subject: [PATCH] drm/hisilicon/hibmc: Add drm_dev_set_unique() for v4.4
Patch-mainline: Never, driver not available in older kernels
References: fate#322061

v4.4 DRM requires a unique name to be set. This was not required in the
upstream driver so was never implemented.

Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c b/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c
index 6974ffaf34e2..7e5fa282d95c 100644
--- a/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c
+++ b/drivers/gpu/drm/hisilicon/hibmc/hibmc_drm_drv.c
@@ -397,6 +397,10 @@ static int hibmc_pci_probe(struct pci_dev *pdev,
 		goto err_disable;
 	}
 
+	ret = drm_dev_set_unique(dev, dev_name(&pdev->dev));
+	if (ret < 0)
+		goto err_unload;
+
 	ret = drm_dev_register(dev, 0);
 	if (ret) {
 		DRM_ERROR("failed to register drv for userspace access: %d\n",
-- 
2.12.2

